src/test/scala/org/ergoplatform/appkit/ergotool/AssetsAtomicExchange/ListMatchingContractsSpec.scala

```diff
@@ -1,59 +1,122 @@
     matchingContracts(Seq(MockInputBox(0L)), Seq(MockInputBox(0L))) shouldBe empty
+  }
 
-    val anyAddress = Address.create("9f4QF8AD1nQ3nJahQVkMj8hFSVVzVom77b52JU7EW71Zexg6N8v")
-    val sellerContract = SellerContract.contractInstance(0,0L, anyAddress.getPublicKey)
-    val token = new ErgoToken("21f84cf457802e66fb5930fb5d45fbe955933dc16a72089bf8980797f24e2fa1", 0L)
+  property("empty list (only a seller contract is given)") {
+    // seller contract with a token
+    val anyAddress = testnetAddressGen.sample.get
+    val sellerContract = SellerContract.contractInstance(0,0L, anyAddress)
+    val token = new ErgoToken(ergoIdGen.sample.get, 1L)
     val sellerBox = MockInputBox(ergoIdGen.sample.get, 1L, sellerContract.getErgoTree, Seq(token))
 
     matchingContracts(Seq(sellerBox), Seq(MockInputBox(0L))) shouldBe empty
   }
 
+  property("empty list (no matching token)") {
+    val sellerContract = SellerContract.contractInstance(0,0L,
+      testnetAddressGen.sample.get)
+    val sellerBox = MockInputBox(ergoIdGen.sample.get, 1L, sellerContract.getErgoTree,
+      Seq(new ErgoToken(ergoIdGen.sample.get, 1L)))
+
+    val buyerContract = BuyerContract.contractInstance(0,
+      new ErgoToken(ergoIdGen.sample.get, 1L), testnetAddressGen.sample.get)
+    val buyerBox = MockInputBox(ergoIdGen.sample.get, 1L, buyerContract.getErgoTree)
+
+    matchingContracts(Seq(sellerBox), Seq(buyerBox)) shouldBe empty
+  }
+
+  property("empty list (no matching token count)") {
+    val tokenPrice = 10L
+    val sellerContract = SellerContract.contractInstance(0, tokenPrice,
+      testnetAddressGen.sample.get)
+    val tokenId = ergoIdGen.sample.get
+    val sellerBox = MockInputBox(ergoIdGen.sample.get, 1L, sellerContract.getErgoTree,
+      Seq(new ErgoToken(tokenId, 1L)))
+
+    val buyerContract = BuyerContract.contractInstance(0, new ErgoToken(tokenId, 2L),
+      testnetAddressGen.sample.get)
+    val buyerBox = MockInputBox(ergoIdGen.sample.get, tokenPrice + MinFee, buyerContract.getErgoTree)
+
+    matchingContracts(Seq(sellerBox), Seq(buyerBox)) shouldBe empty
+  }
+
+  property("empty list(wrong contracts)") {
+    val sellerContract = SellerContract.contractInstance(0,20L,
+      testnetAddressGen.sample.get)
+    val token = new ErgoToken(ergoIdGen.sample.get, 1L)
+    val sellerBox = MockInputBox(ergoIdGen.sample.get, 1L, sellerContract.getErgoTree,
+      Seq(token))
+
+    val buyerContract = BuyerContract.contractInstance(0, token, testnetAddressGen.sample.get)
+    val buyerBox = MockInputBox(ergoIdGen.sample.get, 20L + MinFee, buyerContract.getErgoTree)
+
+    // swapped parameter places
+    matchingContracts(Seq(buyerBox), Seq(sellerBox)) shouldBe empty
+    // same contract for both parameters
+    matchingContracts(Seq(buyerBox), Seq(buyerBox)) shouldBe empty
+    matchingContracts(Seq(sellerBox), Seq(sellerBox)) shouldBe empty
+  }
+
+  property("empty list (DEX fee < 0)") {
```

Comment:
If DexFee is less than MinFee it is also error. This should also be taken into consideration during matching.

Reply:
Fixed and re-written matching tests with prop-checks. See https://github.com/aslesarenko/ergo-tool/pull/10/commits/1f3594163f7fddfceec90a7a03b5890c4d9c3edb